### PR TITLE
removed blocked TG feature flag from alert display logic

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- The Blocked Triage Group Alert has been tested and approved, so this removes the logic for the feature flag toggle, fully enabling the Blocked Triage Group Alert

### [MHV-52371](https://jira.devops.va.gov/browse/MHV-52371) - Cleanup/Remove feature flag: VAgov_MHV_SM_BlockedTriageGroup_1.0 ###

> Cleanup/Remove feature flag:  VAgov_MHV_SM_BlockedTriageGroup_1.0
> 
> User Story: As a Secure Messaging User, I want to remove the feature flag code for all blocked triage group messaging so that functionality is shown and our code is cleaned up when the functionality is on.
> 
> Given:
> When:
> Then:
> 
> Feature Flag Y/N ? N
> Manual Testing Y/N ? Y
> Automated Testing Y/N ? Y
> Accessibility Testing Y/N ? N
> UCD Validation? Y/N N

## Testing done

- Manual testing for each scenario
- updated unit tests that relied on the feature toggle logic

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

VA.gov - Secure Messaging

## Acceptance criteria

> AC1 Feature flag is removed in the code called 'VAgov_MHV_SM_BlockedTriageGroup_1.0
> 
> AC2 Note - all stories related to this should also be attached to fixVersion named the same as in AC1

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by IGallupSoCo** [comment]
It looks like this e2e spec is consistently failing amongst CI runs:
secure-messaging-compose-no-provider.cypress.spec.js

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 1]
```diff
@@ -608,77 +598,51 @@ const ComposeForm = props => {
...
-                      </option>
-                    ))}
-                  </VaSelect>
-                </>
-              )}
+          {recipientsList &&
+            !noAssociations &&
+            !allTriageGroupsBlocked && (
+              <>
+                <VaSelect
```
Comment (micahchiang) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:622]: Can this be updated to v3 by adding the `uswds` prop?

File: src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ const BlockedTriageGroupAlert = props => {
   );
 
   const ehrDataByVhaId = useSelector(
-    state => state.drupalStaticData.vamcEhrData.data.ehrDataByVhaId,
+    state => state.drupalStaticData?.vamcEhrData?.data?.ehrDataByVhaId,
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx:27]: We could technically use this approach as well
```suggestion
import { selectEhrDataByVhaId } from 'platform/site-wide/drupal-static-data/source-files/vamc-ehr/selectors';

      const ehrDataByVhaId = useSelector(selectEhrDataByVhaId);
```

File: src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx [hunk 3]
```diff
@@ -208,17 +202,13 @@ const FolderThreadListView = props => {
       if (threadList?.length === 0) {
         return (
           <>
-            {mhvSecureMessagingBlockedTriageGroup1p0 ? (
+            {!noAssociations &&
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx:205]: is `!noAssociations` addition here intentional? It doesn't seem that we used it here before, perhaps its a new finding?

**Review by alex-morgun-1 — DISMISSED** [review]
Super! Left a suggestion (optional) and a question. Feel free to disregard if not relevant. Otherwise, approved

**Comment by oleksii-morgun-va** [comment]
> It looks like this e2e spec is consistently failing amongst CI runs: secure-messaging-compose-no-provider.cypress.spec.js

@fazilqa are you able to take a look please?

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx (hunk 1 above)
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:622]: @micahchiang is it alright if we pull this work in a separate effort? We are hoping that all `uswds` transition changes are tested properly in its own scope

Comment (micahchiang) [reply src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:622]: You can, but you should be aware that V3 components are becoming opt out only soon, and it'll be your team's responsibility to ensure your application continues to function as normal. [More information in slack](https://dsva.slack.com/archives/C03R5SBELQM/p1707929381167999)

**Review by micahchiang — DISMISSED** [review]

**Review by micahchiang — DISMISSED** [review]

**Review by vsaleem — DISMISSED** [review]
Approved pending result to Oleksii's question -- Looks great!

File: src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx (hunk 3 above)
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/containers/FolderThreadListView.jsx:205]: Yes, this should have been included previously, but it got missed.

File: src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx (hunk 2 above)
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/components/shared/BlockedTriageGroupAlert.jsx]: I like that suggestion, will update code.

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by alex-morgun-1 — APPROVED** [review]
re-approving

**Pull request merged**